src/txn/txn_rollback_to_stable.c

```diff
@@ -1246,9 +1294,10 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
          * 1. The tree is modified.
          * 2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
          * 3. There is no durable timestamp in any checkpoint.
+         * 4. The checkpoint newest txn is greater than snapshot min txn id
          */
         if (S2BT(session)->modified || max_durable_ts > rollback_timestamp || prepared_updates ||
-          !durable_ts_found) {
+          !durable_ts_found || (rollback_txnid > S2C(session)->recovery_ckpt_snap_min)) {
```

Comment:
Can you perform all the transaction id checks only during recovery? In all other cases, we need to worry about transaction ids.

Reply:
@kommiharibabu Updated code to check for transaction ids only during recovery.